### ISSUE TITLE
Why doesn't TensorBoard order objects according to information flow?

### ISSUE THREAD

**Issue by eriophora** [tensorflow/tensorflow#1782]
I would infer that TensorBoard orders objects according to how information flows through the graph, but this doesn't appear to be the case. Namely, considering Inception-v3. The operation makes it clear that, in each conv unit, batch normalization is performed on the output of the convolution layer:

```
conv = tf.nn.conv2d(inputs, weights, [1, stride, stride, 1],
                    padding=padding)
if batch_norm_params is not None:
  with scopes.arg_scope([batch_norm], is_training=is_training,
                        trainable=trainable, restore=restore):
    outputs = batch_norm(conv, **batch_norm_params)
```

i.e., `outputs` is `batch_norm(conv(inputs, ...), ...)`

However, the TensorBoard representation of the graph orders `batch_norm` before the `conv` operation. 

![screenshot from 2016-04-05 11 15 36](https://cloud.githubusercontent.com/assets/6423093/14293082/e17a1776-fb1f-11e5-8073-341d087f11cb.png)

Granted, these don't appear to be connected by a line. However, it *does* order the convolutions correctly:

![screenshot from 2016-04-05 11 17 32](https://cloud.githubusercontent.com/assets/6423093/14293119/031168da-fb20-11e5-9c7a-8f2c6596088f.png)

Is this due to the fact that these nodes are disconnected "given" `Identity...` and are thereby ordered alphabetically? Is there a way to disable this, and have it structure the graph according to the flow of information? Presumably it can disambiguate between the various `Identity...` nodes. Or am I interpreting this situation wrong?

**Comment by dsmilkov** [comment]
Hi eriophora@,

To make sure I understand your question: Are you asking for "weights" to be left of "BatchNorm" inside the conv0 namespace? Left-to-right ordering is meaningless in the graph visualizer. What matters is the bottom-to-top ordering. We leave left-to-right ordering to the layout optimizer to figure it out, which sometimes can swap nodes to reduce edge crossing and we can't control it (even when no edges are shared).

**Comment by michaelisard** [comment]
Closing due to lack of response. Please re-open if you would like this readdressed.
